File: 46-permutations/46-permutations.java
class Solution {
    public List<List<Integer>> ans = new ArrayList<>();
    public List<List<Integer>> permute(int[] nums) {
       List<Integer>que = new ArrayList<>();
       solve(nums,que);
      return ans;
    }
    public void solve(int []nums,List<Integer>que)
    {
        if(que.size()==nums.length)
        {
         ans.add(new ArrayList(que));
            return ;
        }

        for(int i : nums)
        {
            if(que.contains(i)) continue;
            else {
                que.add(i);
                solve(nums,que);
                que.remove(que.size()-1);
            }
            
        
           
        }
    }
}

File: 46-permutations/46-permutations.cpp
class Solution
{
    private:
        vector<vector < int>> ans;
    public:
        void solve(vector<int>&nums,int idx)
        {
            if(idx == nums.size())
            {
                ans.push_back(nums);
            }
            for(int i = idx;i<nums.size();i++)
            {
                swap(nums[idx],nums[i]);
                solve(nums,idx+1);
                swap(nums[idx],nums[i]);
            }
        }
            
        vector<vector < int>> permute(vector<int> &nums) {
            solve(nums,0);
            return ans;
        }
};